src/hokuyo3d.cpp

```diff
@@ -154,44 +156,51 @@ class Hokuyo3dNode
     }
   }
   void cbError(
-      const vssp::Header& header,
-      const std::string& message,
-      const boost::posix_time::ptime& time_read)
+      const vssp::Header &header,
+      const std::string &message,
+      const boost::posix_time::ptime &time_read)
   {
-    ROS_ERROR("%s", message.c_str());
+    RCLCPP_ERROR(this->get_logger(), "%s", message.c_str());
   }
   void cbPing(
-      const vssp::Header& header,
-      const boost::posix_time::ptime& time_read)
+      const vssp::Header &header,
+      const boost::posix_time::ptime &time_read)
   {
+    /*
     const ros::Time now = ros::Time::fromBoost(time_read);
     const ros::Duration delay =
         ((now - time_ping_) - ros::Duration(header.send_time_ms * 0.001 - header.received_time_ms * 0.001)) * 0.5;
     const ros::Time base = time_ping_ + delay - ros::Duration(header.received_time_ms * 0.001);
-
+    */
+    const rclcpp::Time now = this->now();
```

Comment:
`time_read` is a time when the TCP data is arrived. `now()` should have delay.

```suggestion
    const auto t_from_epoch = time_read - boost::posix_time::from_time_t(0);                      
    const auto t_chrono = std::chrono::system_clock::from_time_t(t_from_epoch.total_seconds()) +  
      std::chrono::nanoseconds(t_from_epoch.fractional_seconds() *            
      (1000000000 / t_from_epoch.ticks_per_second()));
    const rclcpp::Time now(t_chrono);
```
(haven't tested and would be better to make it a helper function)